hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/storage/HoodieFileWriterFactory.java

```diff
@@ -45,7 +45,8 @@
       TaskContextSupplier taskContextSupplier) throws IOException {
     final String extension = FSUtils.getFileExtension(path.getName());
     if (PARQUET.getFileExtension().equals(extension)) {
-      return newParquetFileWriter(instantTime, path, config, schema, hoodieTable, taskContextSupplier);
+      return newParquetFileWriter(instantTime, path, config, schema, hoodieTable, taskContextSupplier, config.populateMetaFields(),
```

Comment:
why would we need to pass the same value twice?

Reply:
if you want this control for testing, lets add a new overload for `newParquetFileWriter() `?

Reply:
As I mentioned in notes, last argument decides whether to enable bloom filter or not. In this patch, I am haven't fixed all indexes. But it is possible to fix other index to work for virtual keys. So, added this additional argument whether to enable bloom filter or not. Even for virtual keyed table, if one wants to use regular bloom index, they should be able to.